### DISCUSSION TITLE
How to import Rooms card in HA

### DISCUSSION THREAD

**Discussion by dylanlevymorini** [Tasshack/dreame-vacuum#31] [Q&A] [answered]
Hello, first of all I want to say you it's an awesome project !! I was waiting for it since very long time. Thanks a lot

I really don't know how to import the rooms card part of your documentation. (I've installed custom cards  who are required with this template via HACS).

I copy/past the template on a manual card configuration + provided into it my entity ID but I still have the same error at each time : 
""
Configuration errors detected:

    missed comma between flow collection entries (3:2)
""
screenshot in attachment.
<img width="1357" alt="Screenshot 2022-11-04 at 23 04 48" src="https://user-images.githubusercontent.com/96740910/200081774-d3505802-edc0-4b3a-92c3-bdd59b4d2a8f.png">

Thanks a lot in advance for your help

**Comment by Tasshack** [comment, marked as answer]
You need to use the template editor in the `Developer Tools` for creating the card yaml but you may need to enable the developer tools from settings first.

1- Paste the template on the left text area.
2- Set your vacuum entity id
3- Copy the generated template from the right text area 
4- Paste it into the card configuration.

![template](https://user-images.githubusercontent.com/19640547/200091234-1b5fcebd-32ef-44ed-b688-b1d644f158ec.PNG)

**Comment by dvhub** [comment]
My room cards are missing some of the settings for my L10s Ultra and L10 Pro:

![image](https://user-images.githubusercontent.com/6943588/217643775-02be4368-8394-49e1-8804-c3ae7e6b999e.png)

Do you have any suggestions on how to fix this?  I've confirmed that the entity ID's conform with the template (ex. select.dreame_l10s_ultra_room_7_mop_pad_humidity).

**Reply by Tasshack** [reply]
I have checked the template and the integration for possible reason but I am not sure what is actually wrong with the card also I cannot reproduce the issue on my setup.
Can you check the browser console for possible errors regarding the Mulitple Entity Row?

**Reply by decrepitsvl** [reply]
The same for me, the settings are there but not really visible. One image is edited to show that icons are there
![vacuum2](https://user-images.githubusercontent.com/125564882/233189169-815d1a0c-2ff9-478b-a47e-c9af9bea3546.png)
![vacuum1](https://user-images.githubusercontent.com/125564882/233189172-300859cc-0776-49d5-9079-d99d018b8970.PNG)

**Comment by thalele** [comment]
for me template won't work, btw when i add the card into dashboard and use "Tassahack/dreame-vacuum" and the "Generata rooms configuration" button, now all it works perfectly (Dreame L10s Ultra).

Last thing i don't have for now is a possibility to have a "button" to clean only one room with custom settings for mop etc

**Reply by Tasshack** [reply]
Yes, because the version that supports `dreame_vacuum` of the map card has just released.

**Comment by qdrop17** [comment]
Hi everyone. It's amazing how people reverse engineer these vacuums to integrate them into HA!

I have the same issue: My dreame D10s (dreame.vacuum.r2243) is properly integrated but I'm lacking the map feature too (to control and direct the vacuum). I can display the map though.

Basically this feature here: https://github.com/Tasshack/dreame-vacuum/blob/master/docs/map.md

How can I enable this? I'm using v2.0.0b6.
